lib/services/platform-service.ts

```diff
@@ -351,25 +348,31 @@ export class PlatformService implements IPlatformService {
 		return prepareInfo.changesRequireBuildTime !== buildInfo.prepareTime;
 	}
 
-	public async trackProjectType(): Promise<void> {
+	public async trackProjectType(projectData: IProjectData): Promise<void> {
 		// Track each project once per process.
 		// In long living process, where we may work with multiple projects, we would like to track the information for each of them.
-		if (this.$projectData && (this.$projectData.projectFilePath !== this._trackedProjectFilePath)) {
-			this._trackedProjectFilePath = this.$projectData.projectFilePath;
+		if (projectData && (projectData.projectFilePath !== this._trackedProjectFilePath)) {
+			this._trackedProjectFilePath = projectData.projectFilePath;
 
-			await this.$analyticsService.track("Working with project type", this.$projectData.projectType);
+			await this.$analyticsService.track("Working with project type", projectData.projectType);
 		}
 	}
+	@exportedPromise("platformService")
+	public async build(platform: string, platformBuildOptions: IPlatformBuildData, platformTemplate?: string): Promise<void> {
+		this.$projectData.initializeProjectData(platformBuildOptions.projectDir);
+		await this.preparePlatform(platform, platformBuildOptions, platformTemplate, this.$projectData);
```

Comment:
Given that there are this many changes already I'd personally prefer a separate PR